src/coreclr/jit/importer.cpp

```diff
@@ -10644,10 +10644,10 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 
                     if (isVolatile)
                     {
-                        // Wrap with memory barriers: full-barrier + call + load-barrier
-                        impAppendTree(gtNewMemoryBarrier(), CHECK_SPILL_ALL, impCurStmtDI);
+                        // Wrap with memory barriers: store-barrier + call + load-barrier
+                        impAppendTree(gtNewMemoryBarrier(false, true), CHECK_SPILL_ALL, impCurStmtDI);
```

Comment:
With enum parameter this would look like: `gtNewMemoryBarrier(BARRIER_STORE_ONLY)` 
It would be more self-documenting than `gtNewMemoryBarrier(false, true)`